app/src/main/java/com/example/myapplication/LoginFragment.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.myapplication
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.fragment.app.Fragment
+
+class LoginFragment(val credentialsManager : CredentialsManager) : Fragment() {
+
+    private var listener: EventsListener? = null
+
+    interface EventsListener {
+        fun onRegisterPressed()
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        require(
+            context is EventsListener,
+            { "Activity holding fragment must implement its EventsInterface " }
+        )
+        listener = context
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_create_account, container, false)
```

Comment:
If you're not doing anything with the view, there's no need to create a local val for it
But that's also in realm of own taste imo